mirror-service/src/main/java/org/hyades/client/NvdClient.java

```diff
@@ -0,0 +1,103 @@
+package org.hyades.client;
+
+import io.github.jeremylong.nvdlib.NvdCveApi;
+import io.github.jeremylong.nvdlib.nvd.DefCveItem;
+import org.apache.kafka.streams.processor.api.ContextualProcessor;
+import org.apache.kafka.streams.processor.api.Processor;
+import org.apache.kafka.streams.processor.api.ProcessorContext;
+import org.apache.kafka.streams.processor.api.ProcessorSupplier;
+import org.apache.kafka.streams.processor.api.Record;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.cyclonedx.model.Bom;
+import org.hyades.model.Vulnerability;
+import org.hyades.nvd.NvdMirrorHandler;
+import org.hyades.nvd.NvdToCyclonedxParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+/**
+ * Client for the NVD REST API.
+ */
+public class NvdClient extends ContextualProcessor<String, String, String, Bom>
+        implements ProcessorSupplier<String, String, String, Bom> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NvdClient.class);
+    private static final String LAST_MODIFIED_EPOCH_KEY = "LAST_MODIFIED_EPOCH_KEY";
+    private KeyValueStore<String, Long> store;
+    private StoreBuilder<KeyValueStore<String, Long>> lastModifiedEpochStoreBuilder;
+    private final BiFunction<String, Long, NvdCveApi> cveApiSupplier;
+    private String apiKey;
+
+    Collection<DefCveItem> nvdFeeds;
+
+    public NvdClient(StoreBuilder<KeyValueStore<String, Long>> lastModifiedEpochStoreBuilder,
+                     String apiKey, BiFunction<String, Long, NvdCveApi> cveApiSupplier) {
+        this.apiKey = apiKey;
+        this.nvdFeeds = new ArrayList<>();
+        this.lastModifiedEpochStoreBuilder = lastModifiedEpochStoreBuilder;
+        this.cveApiSupplier = cveApiSupplier;
+    }
+
+    @Override
+    public void init(ProcessorContext context) {
```

Comment:
Minor nit: Please add the generic type indicators here.

```suggestion
    public void init(ProcessorContext<String, Bom> context) {
```